### PR TITLE
NN-3375:

### PR DESCRIPTION
**Pull request by JamieDraperUK**
- Various fixes

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt [hunk 1]
```diff
@@ -91,12 +92,15 @@ class AppointmentService(
 
   @Transactional
   fun createAppointment(createAppointmentSpecification: CreateAppointmentSpecification): CreatedAppointmentDetailsDto {
-    val appointmentsCreated = prisonApiService.createAppointments(makePrisonAppointment(createAppointmentSpecification))
+    val appointmentsCreated =
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt:95]: Minor thing, but shouldn't this be singular `appointmentCreated`

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt [hunk 2]
```diff
@@ -192,19 +192,19 @@ class CourtService(
         VideoLinkBookingResponse.LocationTimeslot(
           locationId = it.eventLocationId,
           startTime = it.startTime,
-          endTime = it.endTime
+          endTime = it.endTime!!
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt:195]: This will throw an NPE if endTime is null

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt [hunk 3]
```diff
@@ -192,19 +192,19 @@ class CourtService(
...
         )
       },
       main = VideoLinkBookingResponse.LocationTimeslot(
         locationId = mainEvent.eventLocationId,
         startTime = mainEvent.startTime,
-        endTime = mainEvent.endTime
+        endTime = mainEvent.endTime!!
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt:201]: See above

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt [hunk 4]
```diff
@@ -192,19 +192,19 @@ class CourtService(
...
       ),
       post = postEvent?.let {
         VideoLinkBookingResponse.LocationTimeslot(
           locationId = it.eventLocationId,
           startTime = it.startTime,
-          endTime = it.endTime
+          endTime = it.endTime!!
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt:207]: See above

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtServiceTest.kt [hunk 5]
```diff
@@ -837,17 +837,17 @@ class CourtServiceTest {
           pre = VideoLinkBookingResponse.LocationTimeslot(
             locationId = preAppointment.eventLocationId,
             startTime = preAppointment.startTime,
-            endTime = preAppointment.endTime
+            endTime = preAppointment.endTime!!
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtServiceTest.kt:840]: Need tests for null endTime

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt (hunk 2 above)
Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt:195]: This shouldn't ever be null as end time is required for Video link bookings. I cleaner way would be to have another model but seemed like over kill.

Comment (sp-watson) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/CourtService.kt:195]: Apparently never null for VLBs

**Review by sp-watson — DISMISSED** [review]

**Review by sp-watson — APPROVED** [review]
